gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/common/MongoFactory.java

```diff
@@ -139,28 +139,27 @@ private ServerSettings buildServerSettings() {
     private SslSettings buildSslSettings() {
         SslSettings.Builder sslBuilder = SslSettings.builder();
 
-        Boolean sslEnabled = readPropertyValue(propertyPrefix + "sslEnabled", Boolean.class);
-
-        if (sslEnabled != null) {
-            sslBuilder.enabled(sslEnabled);
-            if (sslEnabled.booleanValue()) {
-                String keystore = readPropertyValue(propertyPrefix + "keystore", String.class);
-                String keystorePassword = readPropertyValue(propertyPrefix + "keystorePassword", String.class);
-                String keyPassword = readPropertyValue(propertyPrefix + "keyPassword", String.class);
-
-                if (keystore != null) {
-                    try {
-                        SSLContext ctx = SSLContext.getInstance("TLS");
-                        KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
-                        KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
-                        ks.load(new FileInputStream(keystore), keystorePassword.toCharArray());
-                        keyManagerFactory.init(ks, keyPassword.toCharArray());
-                        ctx.init(keyManagerFactory.getKeyManagers(), null, null);
-                        sslBuilder.context(ctx);
-                    } catch (Exception e) {
-                        logger.error(e.getCause().toString());
-                        throw new IllegalStateException("Error creating the keystore for mongodb", e);
-                    }
+        Boolean sslEnabled = readPropertyValue(propertyPrefix + "sslEnabled", Boolean.class, false);
+        sslBuilder.enabled(sslEnabled);
+
+        if (sslEnabled) {
+            String keystore = readPropertyValue(propertyPrefix + "keystore", String.class);
+            String keystorePassword = readPropertyValue(propertyPrefix + "keystorePassword", String.class);
+            String keyPassword = readPropertyValue(propertyPrefix + "keyPassword", String.class);
+            String tlsProtocol = readPropertyValue(propertyPrefix + "tlsProtocol", String.class, "TLSv1.2");
```

Comment:
Is it not better to use SSLContext.getDefault() as the default value instead of fixing it ?
TLS 1.2 is already the default value since JDK 7 so we are good to go with it, and it protect us from using something which is not existing (for any reason...)

WDYT ?

Reply:
It would also ensure coherence depending if you are configuring the URI vs the properties.

Reply:
I was worried to end in a situation where the selected protocol would be "too new" and not handled on the mongo side but: 
 - it looks like even the old mongo 3.6 is supporting `TLSv1.3`
 - at the end, the user would be able to use the config to set up a support version of the protocol
 
 If it's ok for you I will do the update and force push on this branch

Reply:
I do agree with your point, but at the same time, it appears that SSLContext.getInstance("TLS") already return a tls supported by the jvm, so most probably tls1.2 or tls 1.3 depending on the jvm version.
So using the getDefault is not a big issue.

Btw, regarding the behavior of SSLContext.getInstance("TLS")  I'm curious to see what is the vulnerability here. Sounds like a false positive to me....